waveparser_test.go

```diff
@@ -0,0 +1,102 @@
+package waveparser
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+type wavExpectedHeader struct {
+	RIFFHeader struct {
+		Ident     string
+		ChunkSize uint32
+		FileType  string
+	}
+	RIFFChunkFmt   riffChunkFmt
+	FirstSamplePos uint32
+	DataBlockSize  uint32
+}
+
+func checkerr(t *testing.T, err error) {
+	t.Helper()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func testParseWAV(t *testing.T, filename string) {
+	r, err := os.Open(filename)
+	checkerr(t, err)
+
+	hdr, hdrerr := parseHeader(r)
+
+	ext := filepath.Ext(filename)
+	noext := string(filename[0 : len(filename)-len(ext)])
```

Comment:
This could make it easier to strip the extension (a suffix):

https://golang.org/pkg/strings/#TrimSuffix